website/server/libs/chatReporting/inboxChatReporter.js

```diff
@@ -0,0 +1,105 @@
+import nconf from 'nconf';
+import { model as User } from '../../models/user';
+
+import ChatReporter from './chatReporter';
+import {
+  BadRequest,
+} from '../errors';
+import { getGroupUrl, sendTxn } from '../email';
+import slack from '../slack';
+import apiError from '../apiError';
+
+import * as inboxLib from '../inbox';
+
+const FLAG_REPORT_EMAILS = nconf.get('FLAG_REPORT_EMAIL').split(',').map((email) => {
+  return { email, canSend: true };
+});
+
+export default class InboxChatReporter extends ChatReporter {
+  constructor (req, res) {
+    super(req, res);
+
+    this.user = res.locals.user;
+    this.inboxUser = res.locals.user;
+  }
+
+  async validate () {
+    this.req.checkParams('messageId', apiError('messageIdRequired')).notEmpty();
+
+    let validationErrors = this.req.validationErrors();
+    if (validationErrors) throw validationErrors;
+
+    if (this.user.contributor.admin && this.req.query.userId) {
+      this.inboxUser = await User.findOne({_id: this.req.query.userId});
+    }
+
+    const message = await inboxLib.getUserInboxMessage(this.inboxUser, this.req.params.messageId);
+    if (!message) throw new BadRequest(this.res.t('messageGroupChatNotFound'));
+
+    const userComment = this.req.body.comment;
+
+    return {message, userComment};
+  }
+
+  async notify (message, userComment) {
+    const group = {
+      type: 'private messages',
+      name: 'N/A',
+      _id: 'N/A',
+    };
+
+    await super.notify(group, message);
+
+    const groupUrl = getGroupUrl(group);
+    sendTxn(FLAG_REPORT_EMAILS, 'flag-report-to-mods-with-comments', this.emailVariables.concat([
+      {name: 'GROUP_NAME', content: group.name},
+      {name: 'GROUP_TYPE', content: group.type},
+      {name: 'GROUP_ID', content: group._id},
+      {name: 'GROUP_URL', content: groupUrl || 'N/A'},
+      {name: 'REPORTER_COMMENT', content: userComment || ''},
+    ]));
```

Comment:
The emails that mods receive aren't always correct. For example, in the email body below, the flagged PM was in "A Little Yellow Spider"'s inbox, "A Little Yellow Spider" both wrote and reported the PM, and "Author" should therefore be "A Little Yellow Spider" but it shows the person that the PM was sent to.

Ideally the email would say:
- who wrote the PM
- who it was sent to
- who reported it

We can create a new email template in Mandril for this if needed.

Also, the "View group" link at the bottom of the email is non-functional, but that's a really minor point. If removing it is not trivial, just leave it there.

The emails are correct when you're flagging a post that someone else sent to you.


```
Offending Content
Tue Nov 27 2018 08:19:31 GMT+0000 (UTC)
alystest: pm from ALittleYellowSpider to Alys. ALittleYellowSpider to flag.

Reporter's Comments
This is ALittleYellowSpider flagging the PM ALittleYellowSpider sent to Alys.

Persons Involved
Author: lady_alys@oldgods.net - Alys - d904bd62-da08-416b-a816-ba797c9ee265
View Profile

Reporter: alice.harris@oldgods.net - A Little Yellow Spider - 3e595299-3d8a-4a10-bfe0-88f555e4aa0c
View Profile

Group Details
Name: N/A
Type: private messages
ID: N/A
View group
```

Reply:
- added `REPORTER_DISPLAYNAME` and `AUTHOR_DISPLAYNAME` to the email-variables (which can be used for the normal chat report email template)

I use the same template for the inbox reporting. This template has only a `message user` and the `reporting user`, so it isn't able to choose the correct text for the email. 

I think it would be better to create a new email template just for private messages
- @Alys can you create it? 
- or @paglias @SabreCat could you give me access to the templates? or are those hosted in this repo too?

Reply:
Yes, I can create a new template. I'll do that on the weekend. They're stored in Mandrill / MailChimp. I'll report back here with the details you'll need to know after I've done it.

Reply:
I've looked through the email template and I'd actually like us to keep using the same one for flagging both PMs and group messages for ease of maintenance of the templates and because I think it will be easy enough to get the sender/recipient details correct with an `if` statement. Even if a new template was used, we'd still need that `if` to correctly distinguish between sender/recipient. I have adjusted the template so it can be used with PMs without affecting its use for group messages.

Here's the changes that are needed in the code:
- _(This should already be correct for both PMs and group messages.)_ The "REPORTER" details (`REPORTER_UUID`, etc) should always be the person doing the flagging.
- [ ] For private messages, the "AUTHOR" (`AUTHOR_UUID`, etc) should always be the person who sent the message. That can be determined by looking at `msg.sent` and then using either `msg.uuid` (if `msg.sent` is `false`) or `msg.ownerId`/`this.inboxUser` (if `msg.sent` is `true`)
- _(This should already be correct.)_ For group messages, the "AUTHOR" should always be the person who sent the message.
- [ ] For private messages, `AUTHOR_USERNAME` should be changed to contain `user.auth.local.username`. For group messages, it's desirable that the same change be made but not essential (we can change it in a later PR).
- [ ] For private messages, `AUTHOR_DISPLAY_NAME` should be added and should contain `user.profile.name`. For group messages, it's desirable that the same addition be made but not essential.
- [ ] For private messages, new parameters will be needed: `RECIPIENT_EMAIL`, `RECIPIENT_DISPLAY_NAME` (`user.profile.name`), `RECIPIENT_USERNAME` (`user.auth.local.username`), `RECIPIENT_UUID`. They should always indicate the person who received the message.
- _(This is already correct.)_ For group messages, the "RECIPIENT" fields can be left undefined - they do not need to be in the code at all. The email template can handle that.
- Ignore this comment from above, it's not worth fixing: _Also, the "View group" link at the bottom of the email is non-functional, but that's a really minor point. If removing it is not trivial, just leave it there._

Reply:
I'm not certain all of these parameters are being passed through to the email template. For example, I can't see `RECIPIENT_DISPLAY_NAME` in the code, and it's blank in the emails that are sent. The problem might be that "_DISPLAYNAME" is used in the code instead of "_DISPLAY_NAME". It's probably worth fixing just that and then getting it pushed to delta and I'll test again.

NB we use USERNAME with no  underscore and DISPLAY_NAME with an underscore because that most closely mimics how they're written in user-visible text ("Username" and "Display Name").